src/main/java/com/ebivariation/contigalias/controller/AdminController.java

```diff
@@ -0,0 +1,53 @@
+package com.ebivariation.contigalias.controller;
+
+import com.ebivariation.contigalias.entities.AssemblyEntity;
+import com.ebivariation.contigalias.service.AssemblyService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+@RequestMapping("contig-alias-admin")
+@RestController
+public class AdminController {
+
+    private final AssemblyService service;
+
+    public AdminController(AssemblyService service) {
+        this.service = service;
+    }
+
+    @GetMapping(value = "assemblies/{accession}")
+    public Optional<AssemblyEntity> getAssemblyOrFetchByAccession(@PathVariable String accession) throws IOException {
+        return service.getAssemblyOrFetchByAccession(accession);
+    }
+
+    @PostMapping(value = "assemblies/{accession}")
```

Comment:
They are kinda idempotent but only either because thrown exceptions might not always make it to the http response. So for `fetchAndInsertAssemblyByAccession(@PathVariable String accession)` the user does get an exception but on `fetchAndInsertAssemblyByAccession(@RequestBody Optional<List<String>> accessions` it doesn't because of 2 reasons: 

1. Execution is done by thread executor so exception is not propogated back.
2. It takes time to process a list of accessions and by the time an exception is thrown, the http request has already been responded to with a 200 OK.